telemetry_api/lib/telemetry_api/operators.ex

```diff
@@ -29,14 +29,40 @@ defmodule TelemetryApi.Operators do
   ## Examples
 
       iex> get_operator("some_address"})
-      %Operator{}
+      {:ok, %Operator{}}
 
       iex> get_operator("non_existent_address")
-      nil
-
+      {:error, :not_found, "Operator not found for address: non_existent_address"}
   """
   def get_operator(address) do
-    Repo.get(Operator, address)
+    case Repo.get(Operator, address) do
+      nil ->
+        IO.inspect("Operator not found for address: #{address}")
+        {:error, :not_found, "Operator not found for address: #{address}"}
+
+      operator ->
+        {:ok, operator}
+    end
+  end
+
+  @doc """
+  Get a single operator by operator id.
+
+  ## Examples
+
+      iex> get_operator_by_id("some_id")
+      {:ok, %Operator{}}
+
+      iex> get_operator_by_id("non_existent_id")
+      {:error, :not_found, "Operator not found for id: non_existent_id"}
+  """
+  def get_operator_by_id(id) do
```

Comment:
I recommend overloading the `get_operator()` method, so if you send an address it will call `get_operator(address)`, and if you send an id it will call `get_operator(id)`.

You can also do this by meking `get_operator` receive a struct, and you send the parameters like `get_operator(%{id: id})` or `get_operator(%{address: address})` .

You could even receive an `Operator` struct as a param so you are restricted by sending 1 (or n) parameters that already exist in the `Operator` structure.

Reply:
Changed to the map version:

```Elixir
get_operator(%{address: address})
```

If I use an `Operator` struct, the function will match any `%Operator{}` because it automatically assigns nil to all fields internally.